### ISSUE TITLE
Remove 1 audio track from a dual audio mkv file causes audio to become out of sync

### ISSUE THREAD

**Issue by tubularusa** [mbunkus/mkvtoolnix#1909]
I used mkvmergui-gui to remove a Hindu (which is the first) audio track from a dual audio mkv file.  I want the English audio only.  This process causes audio to become out of sync and happens in the latest version of the software.  I've also had this same issue  with similar files.  I've uploaded a sample to the FTP site, ncom.mkv.

**Comment by mbunkus** [comment]
Thank you very much, both for reporting it here, too, and for the sample file.

**Comment by tubularusa** [comment]
I have some additional information that may help.  It appears the audio tracks in the original file are MP3.  I found this out after just extracting the tracks using the mkvextract gui.  I opened the mp3 file up in another program I use called foobar2000, and converted this to AAC using the Nero codec.  I then used the mkvtoolnix-gui to open the ORIGINAL video file with the dual audio.  I then de-selected both audio tracks.  I then added the converted AAC audio file (m4a file) and saved this.  The outputted video plays fine with no sync issues.  I'm guessing the software has an issue with the mp3 audio that is used in some downloaded video files.  Maybe there can be a workaround?

**Comment by mbunkus** [comment]
The issue is with the timestamps of track 3 (the second audio track) in the source file. At timestamp 00:00:24.842 the timestamps suddenly go backwards to 00:00:24.790. There are other places where this happens, too. So far mkvmerge ignores timestamps for MP3 tracks if they go backwards; instead it assigns new, consecutive timestamps.

Track 2 (the first audio track) does not have these issue with its timestamps; they're monotic increasing.

In short: the audio track 3 contains more data than it assigns timestamps for. Players seem to cope with that by slowing down the video in those places.

I don't know why extraction & re-encoding seems to fix the issue for you. Maybe you've experimented with the first audio track, and that one didn't have a problem to begin with.

I'll probably modify mkvmerge to accept timestamps going backwards like that for MP3 audio tracks.

**Comment by tubularusa** [comment]
Thanks mosu!  As I didn't create this file, I can't explain why the
timestamps are like that.  I've had several dual-audio downloaded videos in
the past year that have exhibited this same behavior.  I didn't do any
testing with the first audio track as it is in Hindu.  I'm guessing the
secondary English audio track must not be a priority for whoever creates
these, so maybe they try to keep the file size down by cutting corners in
the compression/bitrate settings, causing this issue.

**Comment by tubularusa** [comment]
Thanks!  Is this something I can look forward to in the next download
release for windows?

**Comment by mbunkus** [comment]
Sure. I'll likely create new pre-builds for Windows later today, too.

**Comment by mbunkus** [comment]
New [pre-builds for Windows](https://mkvtoolnix.download/windows/pre/) are up. Build numbers 01439 and higher contain the fix.

**Comment by tubularusa** [comment]
Thanks a lot!  Very impressive software engineering!

**Comment by mbunkus** [comment]
You're welcome.
